se_services/zephyr/src/se_service.c

```diff
@@ -1184,6 +1185,44 @@ int se_service_boot_reset_cpu(uint32_t cpu_id)
 	return 0;
 }
 
+int se_service_process_toc_entry(const char *image_id)
+{
+	int err, resp_err = -1;
+
+	if (!image_id) {
+		LOG_ERR("Invalid argument\n");
+		return -EINVAL;
+	}
+
+	if (k_mutex_lock(&svc_mutex, K_MSEC(MUTEX_TIMEOUT))) {
```

Comment:
You must always clear errno before you make a function call which might set it.
```suggestion
	errno = 0;
	if (k_mutex_lock(&svc_mutex, K_MSEC(MUTEX_TIMEOUT))) {
```

Reply:
Just noted that the checking the return value of `k_mutex_lock` (which does not set `errno`) is broken throughout the file. I have pushed another patch to fix this (together with a patch to fix potential data corruption) before introducing process_toc_entry service.